internal/overlord/servstate/handlers.go

```diff
@@ -134,6 +150,13 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 	go func() {
 		active.err = cmd.Wait()
 		_ = active.logBuffer.Close()
+		releasePlan, err := m.acquirePlan()
+		if err == nil {
+			if m.services[req.Name].cmd == cmd {
```

Comment:
What does this `if` statement check for? (I realize it was there already below.)

Reply:
acquirePlan wouldn't actually fail here, since the plan is loaded already. I guess I can make this more future proof by relying on the mutex directly incase acquirePlan implementation changes.